docs/source/notes/serialization.rst

```diff
@@ -2,90 +2,337 @@
 Serialization semantics
 =======================
 
-Storage sharing is preserved in serialization
+This note describes how you can save and load PyTorch tensors and module states
+in Python, and how to serialize Python modules so they can be loaded in C++.
+
+.. contents:: Table of Contents
+
+Saving and loading tensors
+--------------------------
+
+.. _saving-loading-tensors:
+
+:func:`torch.save` and :func:`torch.load` let you easily save and load tensors:
+
+::
+
+    >>> t = torch.tensor([1., 2.])
+    >>> torch.save(t, 'tensor.pt')
+    >>> torch.load('tensor.pt')
+    tensor([1., 2.])
+
+By convention, PyTorch files are typically written with a ‘.pt’ or ‘.pth’ extension.
+
+:func:`torch.save` and :func:`torch.load` use Python’s pickle by default,
+so you can also save multiple tensors as part of Python objects like tuples,
+lists, and dicts:
+
+::
+
+    >>> d = {'a': torch.tensor([1., 2.]), 'b': torch.tensor([3., 4.])}
+    >>> torch.save(d, 'tensor_dict.pt')
+    >>> torch.load('tensor_dict.pt')
+    {'a': tensor([1., 2.]), 'b': tensor([3., 4.])}
+
+Custom data structures that include PyTorch tensors can also be saved if the
+data structure is pickle-able.
+
+Saving and loading tensors preserves views
 ---------------------------------------------
 
 .. _preserve-storage-sharing:
 
-PyTorch saves the underlying storages so that tensors sharing the same storage before :func:`torch.save`
-will still share storage after :func:`torch.load`.
+Saving tensors preserves their view relationships:
+
+::
+
+    >>> numbers = torch.arange(1, 10)
+    >>> evens = numbers[1::2]
+    >>> torch.save([numbers, evens], 'tensors.pt')
+    >>> loaded_numbers, loaded_evens = torch.load('tensors.pt')
+    >>> loaded_evens *= 2
+    >>> loaded_numbers
+    tensor([ 1,  4,  3,  8,  5, 12,  7, 16,  9])
+
+Behind the scenes, these tensors share the same "storage." See
+`Tensor Views <https://pytorch.org/docs/master/tensor_view.html>`_ for more
+on views and storages.
+
+When PyTorch saves one or more tensors it saves their storages and tensor
+metadata separately. This is an implementation detail that may change in the
+future, but it typically saves space and lets PyTorch easily
+reconstruct the view relationships between the loaded tensors. In the above
+snippet, for example, only a single storage is written to 'tensors.pt'.
+
+In some cases, however, saving storages may be unecessary and create
```

Comment:
nit: unnecessary, although the wording is a little strange (you have to save *something*), maybe say "saving the entire storage may be unnecessary/wasteful"?

Reply:
Yes, that's better.